### PR TITLE
Add margin_rank_loss_op to python

### PR DESCRIPTION
**Pull request by velconia**

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5827,6 +5828,47 @@ def rank_loss(label, left, right, name=None):
     return out
 
 
+def margin_rank_loss(label, left, right, margin=0.1, name=None):
+    """
+    **Margin Rank loss layer for rank problem**
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: This doesn't seems to explain what does this operator do?

**Review by panyx0718 — COMMENTED** [review]
Is there a test for this operator?

**Comment by velconia** [comment]
Yes, [here](https://github.com/PaddlePaddle/Paddle/blob/0abfbd1c41e6d558f76252854d4d78bef581b720/python/paddle/fluid/tests/unittests/test_margin_rank_loss_op.py) is the unit test

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (velconia) [reply python/paddle/fluid/layers/nn.py]: I enriched the doc in next commit

**Review by panyx0718 — DISMISSED** [review]

**Comment by panyx0718** [comment]
you might need to update wiki

**Comment by velconia** [comment]
The wiki was updated now.

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
     return out
 
 
+def margin_rank_loss(label, left, right, margin=0.1, name=None):
+    """
+    Margin Rank loss layer for rank problem, which comparing left value and right value be passed in.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: ` Margin Rank loss layer for rank problem` -> ` Margin Ranking Loss Layer for ranking problem`
`comparing` -> `compares`
`value` -> `score`
`be passed in` -> `passed in`

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+    The rank loss can be defined as below equation:
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `rank` -> `ranking`
`below` -> `following`

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+
+    .. math::
+
+        rank\_loss &= max(0, -label * (left - right) + margin)
+
+    Args:
+       label (Variable): Indicats whether left higher than (right + margin) or not.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `Indicates whether the left is ranked higher than the right or not. `

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+       left (Variable): rank score for left.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `rank` -> `Ranking`

File: python/paddle/fluid/layers/nn.py [hunk 6]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+       right (Variable): rank score for right.
+       margin (float): Indicates the margin to be added to right
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `Indicates the margin to be added to right` -> `Indicates the given margin`

File: python/paddle/fluid/layers/nn.py [hunk 7]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+       name (str|None): A name for this layer (optional). If set None, the layer
+                       will be named automatically.
+    Returns:
+       list: The value of rank loss.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `list` -> `Variable`
`rank` -> `ranking`

File: python/paddle/fluid/layers/nn.py [hunk 8]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+    Raises:
+       ValueError: Any of label, left, and right is not a variable.
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `variable` -> `Variable`

File: python/paddle/fluid/layers/nn.py [hunk 9]
```diff
@@ -5827,6 +5828,53 @@ def rank_loss(label, left, right, name=None):
...
+           out = fluid.layers.margin_rank_loss(label, left, right)
+    """
+    helper = LayerHelper('margin_rank_loss', **locals())
+    if not (isinstance(label, Variable)):
+        raise ValueError("The label should be a Variable")
+    if not (isinstance(left, Variable)):
+        raise ValueError("The left should be a Variable")
+    if not (isinstance(right, Variable)):
+        raise ValueError("The right should be a Variable")
+    out = helper.create_tmp_variable("float32")
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: `dtype`

**Review by kuke — COMMENTED** [review]
Please refine the document

File: python/paddle/fluid/layers/nn.py (hunk 8 above)
Comment (velconia) [reply python/paddle/fluid/layers/nn.py]: 赞

**Review by kuke — APPROVED** [review]
LGTM

**Review by panyx0718 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**